data/src/main/java/org/apache/iceberg/data/avro/IcebergDecoder.java

```diff
@@ -166,7 +166,7 @@ public D decode(InputStream stream, D reuse) throws IOException {
      * @param writeSchema the schema used to decode buffers
      */
     private RawDecoder(org.apache.iceberg.Schema readSchema, org.apache.avro.Schema writeSchema) {
-      this.reader = new ProjectionDatumReader<>(DataReader::create, readSchema, ImmutableMap.of());
+      this.reader = new ProjectionDatumReader<>(DataReader::create, readSchema, ImmutableMap.of(), null);
```

Comment:
It would be nice to support name mapping in the message decoder as well. This is fine for now, but could you open an issue to track adding support for name mapping for Iceberg generic single-message encoding?

Reply:
Added #500